sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -174,7 +166,11 @@ rom_error_t shutdown_init(lifecycle_state_t lc_state) {
   return error;
 }
 
-uint32_t shutdown_redact(rom_error_t reason, shutdown_error_redact_t severity) {
+/**
+ * Implementation of `shutdown_redact` that is guaranteed to be inlined.
```

Comment:
So, given this is a `static` function, what is the reason for guaranteed inlining here? Probably worth documenting, since it's definitely not obvious.

Reply:
Good point.

The reason is that these functions are called from `shutdown_finalize` which shouldn't perform any function calls since they present a way to 'escape' the shutdown sequence. In particular we probably don't want to jump out of the `.shutdown` section although we currently do because of the `base_printf` call.

Reply:
I've added a note to that effect.